bitcoin/src/bip32.rs

```diff
@@ -125,6 +125,39 @@ pub enum ChildNumber {
         index: u32,
     },
 }
+
+/// NormalChildNumber is a ChildNumber that is not hardened
+/// made only to be used internally in `derive_pub()` and `NormalChildIterator`
+#[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
+pub struct NormalChildNumber(pub u32);
+
+/// A iterator over the children of a DerivationPath trying to convert them into NormalChildNumber
+/// made only to be used internally in `derive_pub()`
+pub struct NormalChildIterator {
+    base: Vec<NormalChildNumber>,
+}
+impl Into<NormalChildIterator> for DerivationPath {
```

Comment:
This is has multiple problems. It should be fallible, it should be `TryFrom` rather than `TryInto` and it shouldn't collect.